Description: [PATCH] conversations: update the G keys even if the message has no
 threadid
Author: Bron Gondwana <brong@fastmail.fm>
Forwarded: no
Reviewed-By: Xavier Guimard <yadd@debian.org>
Last-Update: 2020-02-10

---
 imap/conversations.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

--- a/imap/conversations.c
+++ b/imap/conversations.c
@@ -2353,6 +2353,21 @@
     const struct index_record *record = new ? new : old;
     if (!record) return 0;
 
+    // always update the GUID information first, as it's used for search
+    // even if conversations have not been set on this email
+    if (new) {
+        if (!old) {
+            r = conversations_set_guid(cstate, mailbox, new, /*add*/1);
+            if (r) return r;
+        }
+    }
+    else {
+        if (old) {
+            r = conversations_set_guid(cstate, mailbox, old, /*add*/0);
+            if (r) return r;
+        }
+    }
+
     if (new && !old && allowrenumber) {
         /* add the conversation */
         r = mailbox_cacherecord(mailbox, new); /* make sure it's loaded */
